server/project.go

```diff
@@ -581,40 +588,46 @@ func (s *Server) composeProjectByID(ctx context.Context, id int) (*api.Project,
 	projectFind := &api.ProjectFind{
 		ID: &id,
 	}
-	project, err := s.ProjectService.FindProject(ctx, projectFind)
+	projectPlain, err := s.ProjectService.FindProject(ctx, projectFind)
 	if err != nil {
 		return nil, err
 	}
-	if project == nil {
+	if projectPlain == nil {
 		return nil, &common.Error{Code: common.NotFound, Err: fmt.Errorf("project ID not found: %d", id)}
 	}
 
-	if err := s.composeProjectRelationship(ctx, project); err != nil {
+	project, err := s.composeProjectRelationship(ctx, projectPlain)
+	if err != nil {
 		return nil, err
 	}
 
 	return project, nil
 }
 
-func (s *Server) composeProjectRelationship(ctx context.Context, project *api.Project) error {
+func (s *Server) composeProjectRelationship(ctx context.Context, projectPlain *api.ProjectPlain) (*api.Project, error) {
 	var err error
 
+	project := &api.Project{}
+	if err := copier.Copy(project, projectPlain); err != nil {
```

Comment:
I might prefer we do our own copy & pasting for each fields. Alternatively, we might be able to use golang embedding struct.

Reply:
> Alternatively, we might be able to use golang embedding struct.

I have tried but failed to make `jsonapi` work with the embedding. Any ideas?

If embedding is not working with `jsonapi`, I am OK with manual copying.

@d-bytebase

Reply:
Yes, let's copy fields in a more explicit way.